File: TP1 - C++/data/models/point.cpp
#include "point.hpp"

Point::Point(string id, string date, string observation):
    id(id),
    date(date),
    observation(observation){};

Point::~Point(){};

string Point::getId(){
    return this->id;
}

string Point::getDate(){
    return this->date;
}

string Point::getObservation(){
    return this->observation;
}

void Point::setDate(string date){
    this->date = date;
}

void Point::setObservation(string observation){
    this->observation = observation;
}

File: TP2 - Java/src/domain/entities/Point.java
package domain.entities;

public class Point {
  private final String id;
  private String date;
  private String observation;

  public Point(String id, String date, String observation) {
    this.id = id;
    if (date == null) {
      this.date = "";
    } else {
      this.date = date;
    }
    if (observation == null) {
      this.observation = "";
    } else {
      this.observation = observation;
    }
  }

  public String getId() {
    return id;
  }

  public String getDate() {
    return date;
  }

  public void setDate(String date) {
    this.date = date;
  }

  public String getObservation() {
    return observation;
  }

  public void setObservation(String observation) {
    this.observation = observation;
  }
}
